src/python/pants/jvm/goals/coursier.py

```diff
@@ -123,55 +124,23 @@ async def coursier_generate_lockfile(
     )
 
     resolved_lockfile = await Get(
-        CoursierResolvedLockfile,
-        ArtifactRequirements,
-        artifact_requirements,
+        CoursierResolvedLockfile, ArtifactRequirements, artifact_requirements
     )
     resolved_lockfile_json = resolved_lockfile.to_json()
-
-    lockfile_path = jvm.options.resolves[request.resolve]
-
-    # Materialise the existing lockfile, and check for changes. We don't want to re-write
-    # identical lockfiles
```

Comment:
We were duplicating the two `if` statement bodies. I inverted the conditional to make this all shorter and simpler.